### PR TITLE
Make the CLA check configurable and enable it for Dotty

### PR DESCRIPTION
**Pull request by smarter**

### CODE REVIEW

**Comment by SethTisue** [comment]
I tried to deploy this to test, but got bogged down in breakage and under-documentation. I want to sort it out and at least produce documentation of the current state of things, but it'll have to wait until next week.

**Comment by smarter** [comment]
OK, thanks for looking into this!

File: core/src/main/scala/scabot/core/Configuration.scala [hunk 1]
```diff
@@ -63,9 +63,10 @@ trait Configuration extends Core {
         repo  <- configString(c.get("repo"))
         branches  <- configStringList(c.get("branches"))
         lastCommitOnly <- configBoolean(c.get("lastCommitOnly")) orElse Some(false) // default for scala/scala
+        checkCLA <- configBoolean(c.get("checkCLA")) orElse Some(true) // default for scala/scala
```
Comment (adriaanm) [inline core/src/main/scala/scabot/core/Configuration.scala]: shouldn't the default be `false`, for backwards compat? We'll also need to update scala-jenkins-infra.

**Comment by adriaanm** [comment]
Let's do this, but defer until we have a block of time to do other needed housekeeping as well. We'll also need to update scala-jenkins-infra.

**Comment by adriaanm** [comment]
infra updated -- note that when there's no checkCLA clause in the config we NPE:
```
         at scabot.core.Configuration$class.configBoolean$1(Configuration.scala:32)
```

**Comment by smarter** [comment]
Thanks!

**Pull request merged**